src/main/java/org/sagebionetworks/web/client/widget/entity/controller/ProvenanceEditorWidget.java

```diff
@@ -0,0 +1,216 @@
+				List<ProvenanceEntry> executedEntries = new LinkedList<ProvenanceEntry>();
+				for (Used provEntry: allProvenance) {    
+					ProvenanceEntry toAdd;
+					if (provEntry instanceof UsedEntity) {
+						Reference ref = ((UsedEntity)provEntry).getReference();
+						toAdd = ginInjector.getEntityRefEntry();
+						String entityId = ref.getTargetId();
+						Long version = ref.getTargetVersionNumber();
+						((EntityRefProvEntryView)toAdd).configure(entityId, version != null ? version.toString() : null);
+						toAdd.setAnchorTarget(DisplayUtils.getSynapseHistoryToken(entityId, version));
+					} else {
+						UsedURL usedURL = (UsedURL) provEntry;
+						toAdd = ginInjector.getURLEntry();
+						String name = usedURL.getName();
+						String url = usedURL.getUrl();
+						((URLProvEntryView)toAdd).configure(name != null && !name.trim().isEmpty() ? name : usedURL.getUrl(), url);
+						toAdd.setAnchorTarget(url);
+					}
+					
+					if (provEntry.getWasExecuted()) {
+						executedEntries.add(toAdd);
+					} else {
+						usedEntries.add(toAdd);
+					}
+				}
+				usedProvenanceList.configure(usedEntries);
+				executedProvenanceList.configure(executedEntries);
+			}
+		});
+	}
+	
+	@Override
+	public Widget asWidget() {
+		return view.asWidget();
+	}
+	
+	public void show() {
+		clear();
+		view.show();
+	}
+	
+	public void hide() {
+		view.hide();
+	}
+	
+	@Override
+	public void clear() {
+		usedProvenanceList.clear();
+		executedProvenanceList.clear();
+		synAlert.clear();
+		view.clear();
+	}
+
+	@Override
+	public void onSave() {
+		activity.setName(view.getName());
+		activity.setDescription(view.getDescription());
+		List<ProvenanceEntry> usedEntries = usedProvenanceList.getEntries();
+		List<ProvenanceEntry> executedEntries = executedProvenanceList.getEntries();
+		Set<Used> usedSet = new HashSet<Used>();
+		for (ProvenanceEntry used: usedEntries) {
+			if (used instanceof URLProvEntryView) {
+				URLProvEntryView urlEntry = (URLProvEntryView) used;
+				UsedURL activityURL = new UsedURL();
+				activityURL.setUrl(urlEntry.getURL());
+				activityURL.setName(urlEntry.getTitle());
+				activityURL.setWasExecuted(false);
+				usedSet.add(activityURL);
+			} else {
+				EntityRefProvEntryView entityEntry = (EntityRefProvEntryView) used;
+				UsedEntity activityEntity = new UsedEntity();
+				Reference ref = new Reference();
+				ref.setTargetId(entityEntry.getEntryId());
+				String version = entityEntry.getEntryVersion();
+				// Null version is displayed as "Current"
+				if (version != null && !version.equals("Current")) {
+					ref.setTargetVersionNumber(Long.valueOf(version));
+				}
+				activityEntity.setReference(ref);
+				activityEntity.setWasExecuted(false);
+				usedSet.add(activityEntity);
+			}
+		}
+		for (ProvenanceEntry executed: executedEntries) {
+			if (executed instanceof URLProvEntryView) {
+				URLProvEntryView urlEntry = (URLProvEntryView) executed;
```

Comment:
Lots of duplicate code here.  If the only thing that is different is wasExecuted, feed that in as a parameter (to the shared method that these should be using).